charts/nginx/templates/nginx-configmap.yaml

```
@@ -48,14 +48,16 @@ data:
 
         -- Split the query string into individual parameters
         for param in string.gmatch(queryString, "([^&]+)") do
-          local paramName, paramValue = string.match(param, "(.+)=(.+)")
+          local paramName, paramValue = string.match(param, "(.+)=(.*)") -- Allow empty paramValue
```

We should tighten this regex up a bit. Here's at least one example:
```suggestion
          local paramName, paramValue = string.match(param, "([^=]+)=(.*)") -- Allow empty paramValue
```
I wonder if there is a function we can use to do this, because the regex to reliably capture the parameters could be complex.

---

this seems to add %3D at the end with the new change

---

I think we should not use a regex to do this since there's the built-in function.

---

let me try to optimize the existing code and update the PR tomorrow

---

I'm going to bang on this and try to figure it out. It's not quite working yet. The blank values are still causing a problem.

---

I've been banging on this and I've run into a bunch of edge cases and can't solve them all. I've never done lua before, so that's part of my problem.

One case we should cover is `&foo`

`([^?&]+)=([^&]*)` solves the `&root=` case

---

After digging banging on the code and digging through the API reference <https://github.com/openresty/lua-nginx-module#nginx-api-for-lua> I don't see any good way to do the referer other than regex because there is no function to parse the arguments from URL string. All of the URL parsing functions do not take a URL, they imply the URL that is part of the current request.

---

```
set_by_lua_block $request_uri_masked {
      local args = ngx.req.get_uri_args()
      local paramsToMask = {
        token = true
      }
      local maskedArgs = {}
      for paramName, paramValue in pairs(args) do
        -- Check if the parameter should be masked
        if paramsToMask[paramName] then
          paramValue = "##REDACTED##" -- Replace the value with REDACTED
        end

        if paramName and paramValue then -- Check if paramName and paramValue exist
          table.insert(maskedArgs, paramName .. "=" .. paramValue)
        end
      end
      local maskedQueryString = table.concat(maskedArgs, "&")
      local maskedUrl = ngx.var.uri
      if maskedQueryString ~= "" then
        maskedUrl = maskedUrl .. "?" .. maskedQueryString
      end
      
      return maskedUrl
   
    }
  ```